cat-launcher/src/pages/SettingsPage/components/SettingsForm.tsx

```diff
@@ -0,0 +1,155 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Controller, useForm } from "react-hook-form";
+import * as z from "zod";
+
+import { Button } from "@/components/ui/button";
+import {
+  Field,
+  FieldContent,
+  FieldDescription,
+  FieldError,
+  FieldLabel,
+} from "@/components/ui/field";
+import { Input } from "@/components/ui/input";
+import { toastCL } from "@/lib/utils";
+import { useApplySettings, useSettings } from "../hooks/useSettings";
+import FontSelector from "./FontSelector";
+import ThemeSelector from "./ThemeSelector";
+
+const settingsSchema = z.object({
+  maxBackups: z.number().min(0).max(20),
+  parallelRequests: z.number().min(1).max(16),
+  fontLocation: z.string().nullable(),
+  themeName: z.string().nullable(),
+});
+
+type SettingsFormValues = z.infer<typeof settingsSchema>;
+
+export default function SettingsForm() {
+  const {
+    settings,
+    themes,
+    isLoading: dataLoading,
+  } = useSettings({
+    onSettingsError: (err) =>
+      toastCL("error", "Failed to load settings", err),
+  });
+  const { applySettings, isPending } = useApplySettings({
+    onError: (err) =>
+      toastCL("error", "Failed to apply settings", err),
+  });
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<SettingsFormValues>({
+    resolver: zodResolver(settingsSchema),
+    values: settings
+      ? {
+          maxBackups: settings.max_backups,
+          parallelRequests: settings.parallel_requests,
+          fontLocation: settings.font ?? null,
+          themeName: settings.themeName,
+        }
+      : undefined,
+  });
+
+  const onApply = async (data: SettingsFormValues) => {
+    const selectedThemeObj = themes.find(
+      (t) => t.name === data.themeName,
+    );
+
+    await applySettings({
+      maxBackups: data.maxBackups,
+      parallelRequests: data.parallelRequests,
+      fontLocation: data.fontLocation ?? undefined,
+      themeColors: selectedThemeObj?.colors,
+    });
+
+    toastCL("success", "Settings applied successfully");
```

Comment:
In cat-launcher/src/pages/SettingsPage/components/SettingsForm.tsx around lines
59 to 72, the success toast is shown even when applySettings may fail because
errors can be swallowed by handleSubmit; wrap the await applySettings call in a
try/catch, only call toastCL("success", "Settings applied successfully") after a
successful await, and in the catch call toastCL("error", "...") with a helpful
message (or rethrow the error if the form's onError should handle it) so
failures do not produce a success notification.